src/mango_native_proc.erl

```diff
@@ -163,14 +165,19 @@ get_text_field_values({Props}, TAcc) when is_list(Props) ->
     get_text_field_values_obj(Props, TAcc, []);
 
 get_text_field_values(Values, TAcc) when is_list(Values) ->
+	ArrayLengthFieldEnabled = TAcc#tacc.array_length_field,
     NewPath = ["[]" | TAcc#tacc.path],
-    NewTAcc = TAcc#tacc{path = NewPath},
-    % We bypass make_text_field and directly call make_text_field_name
-    % because the length field name is not part of the path.
-    LengthFieldName = make_text_field_name(NewTAcc#tacc.path, <<"length">>),
-    LengthField = [{LengthFieldName, <<"length">>, length(Values)}],
-    get_text_field_values_arr(Values, NewTAcc, LengthField);
-
+    NewTAcc = TAcc#tacc{array_length_field = ArrayLengthFieldEnabled, path = NewPath},
+	case ArrayLengthFieldEnabled of 
```

Comment:
this variable is more clearly named than the record entry. I suggest `index_array_lengths` and `IndexArrayLengths` would better describe the flag that toggles the indexing of array lengths.